rust/src/connection.rs

```diff
@@ -7,13 +7,7 @@ use anyhow::Result;
 pub async fn keep_alive() -> Result<()> {
     let mut connected = false;
     loop {
-        let channel_details = wallet::get_first_channel_details();
-
-        let is_connection_lost_maybe = channel_details.is_some()
-            && !is_first_channel_usable()
-            && channel_details.unwrap().short_channel_id.is_some();
-
-        if !connected || is_connection_lost_maybe {
+        if !connected || !is_first_channel_usable() {
```

Comment:
If I understand it correctly, `is_first_channel_usable` already checks that the channel exists, so we don't need the first guard.

Also, we think that the channel will only be usable if the `short_channel_id` is present.

Makes sense to me!